src/app/clusters/dishwasher-alarm-server/dishwasher-alarm-server.cpp

```diff
@@ -0,0 +1,186 @@
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogProgress(Zcl, "Dishwasher Alarm: ERR: reading  state, err:0x%x", status);
+        return status;
+    }
+
+    status = Attributes::State::Set(endpoint, newState);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogProgress(Zcl, "Dishwasher Alarm: ERR: writing  state, err:0x%x", status);
+        return status;
+    }
+
+    ChipLogProgress(Zcl, "Dishwasher Alarm: State ep%d value: %" PRIx32 "", endpoint, newState.Raw());
+
+    // Generate Notify event.
+    BitMask<AlarmMap> becameActive;
+    becameActive.Set(newState).Clear(currentState);
+    BitMask<AlarmMap> becameInactive;
+    becameInactive.Set(currentState).Clear(newState);
+    // This field SHALL be a copy of the Mask attribute when this event was generated.
+    BitMask<AlarmMap> mask;
+
+    status = GetMaskValue(endpoint, &mask);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        return status;
+    }
+    SendNotifyEvent(endpoint, becameActive, becameInactive, newState, mask);
+
+    return status;
+}
+
+EmberAfStatus DishwasherAlarmServer::SetSupportedValue(EndpointId endpoint, const BitMask<AlarmMap> supported)
+{
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+    status               = Attributes::Supported::Set(endpoint, supported);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogProgress(Zcl, "Refrigerator Alarm: ERR: writing supported, err:0x%x", status);
+        return status;
+    }
+
+    ChipLogProgress(Zcl, "Refrigerator Alarm: Supported ep%d value: %" PRIx32 "", endpoint, supported.Raw());
+
+    // Whenever there is change in Supported attribute, Mask, State should change accordingly.
+    BitMask<AlarmMap> mask;
+    status = GetMaskValue(endpoint, &mask);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        return status;
+    }
+
+    if (!supported.HasAll(mask))
+    {
+        mask   = supported & mask;
+        status = SetMaskValue(endpoint, mask);
+    }
+    return status;
+}
+
+void DishwasherAlarmServer::SendNotifyEvent(EndpointId endpointId, BitMask<AlarmMap> becameActive, BitMask<AlarmMap> becameInactive,
+                                            BitMask<AlarmMap> newState, BitMask<AlarmMap> mask)
+{
+    Events::Notify::Type event{ .active = becameActive, .inactive = becameInactive, .state = newState, .mask = mask };
+    EventNumber eventNumber;
+    CHIP_ERROR error = LogEvent(event, endpointId, eventNumber);
+    if (CHIP_NO_ERROR != error)
+    {
+        ChipLogError(Zcl, "[Notify] Unable to send notify event: %s [endpointId=%d]", error.AsString(), endpointId);
+    }
+}
+
+/**********************************************************
+ * Callbacks Implementation
+ *********************************************************/
+
+void emberAfDishwasherAlarmClusterServerInitCallback(EndpointId endpoint) {}
+
+void MatterDishwasherAlarmPluginServerInitCallback() {}
```

Comment:
I don't see any handlers for the Reset and ModifyEnabledAlarms commands here.  Do dishwashers not support those?  I don't see anything to that effect in the spec, unlike the Refrigerator Alarm cluster.